### PR TITLE
Explat: Add schema so state is saved in cache

### PR DESCRIPTION
**Pull request by withinboredom**
#### Changes proposed in this Pull Request

* Adds a schema so experiment state will be persisted.

#### Testing instructions

- Go to a page with a running experiment
- Verify you have assignments by looking in `state.experiments` on your console
- Refresh the page
- `nextRefresh` should not update.

### CODE REVIEW

**Review by taggon — APPROVED** [review]
It works! 👍

**Comment by withinboredom** [comment]
I was hoping there would be a way to specify that a value **must** be `true` or `false` in the schema, but this appears to not be possible. This means something could get stuck in the "isLoading" state. This should be very unlikely, as at some point, the experiment had rendered and produced a valid state.

**Comment by taggon** [comment]
@withinboredom Is `type: 'boolean'` not working?

File: client/state/experiments/schema.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export const schema = {
+	type: 'object',
+	additionalProperties: true,
+	properties: {
```
Comment (taggon) [inline client/state/experiments/schema.js:4]: I've tested the following schemas and looks like it works.

```
isLoading: {
    type: 'boolean',
},
```
...
```
variations: {
    type: 'object',
    additionalProperties: false,
    patternProperties: {
        '^[a-z0-9_]+$': {
            oneOf: [
                  { type: 'string' },
                  { type: 'null' },
            ],
        },
    },
},
```

Comment (withinboredom) [reply client/state/experiments/schema.js:4]: Oh nice, thanks!

**Comment by jessie-ross** [comment]
The code makes sense to me, I'll trust you and Taegon testing it.

**Pull request merged**